### DISCUSSION TITLE
problem of years

### DISCUSSION THREAD

**Discussion by Chino70** [Hacker0x01/react-datepicker#4003] [Q&A] [answered]
when only year must be completed and the calendar always opens the previous year

**Comment by aziyatali** [comment]
@Chino70  Please provide more context/code about your problem?

**Comment by Chino70** [comment]
when i enter a single year in the text while the calendar shows previous year, i prefer same year to appear
return (
    <DatePicker
      style={{ width: '100%' }}
      className="form-control"
      locale="es"
      peekNextMonth
      showYearDropdown
      showMonthDropdown
      disabledKeyboardNavigation
      dateFormat="dd/MM/yyyy"
      {...props }
    />
  );
![imagen](https://user-images.githubusercontent.com/24843291/233373414-8b0bf324-928b-4ce3-b1c7-8cce2a93db31.png)

**Comment by aziyatali** [comment, marked as answer]
@Chino70  
You should specify the selected props.
Something like this:
```

const [selectedDate, setSelectedDate] = useState(new Date());

  return (
    <DatePicker
      style={{ width: '100%' }}
      className="form-control"
      locale="es"
      peekNextMonth
      showYearDropdown
      showMonthDropdown
      disabledKeyboardNavigation
      dateFormat="dd/MM/yyyy"
      selected={selectedDate}
      onChange={(date) => setSelectedDate(date)}
      {...props}
    />
  );
```

If it resolves your problem, please specify as resolved

**Comment by Chino70** [comment]
Sorry, I'll give you another code, it's to enter the text

const DatePickerField = (params: DatePickerProps) => {
  const { validate, minDate: min, maxDate: max, blockedMonth, ...props } = params;

return (
    <DatePicker
      type="date"
      validate={(validate) ? validations.concat(validate) : validations}
      previousButtonElement={((blockedMonth) ? "" : "<") }
      nextButtonElement={((blockedMonth) ? "" : ">") }
      {...props}
    />
  );
};
